codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java

```diff
@@ -85,6 +85,7 @@
     private Http2Settings initialSettings = Http2Settings.defaultSettings();
     private Http2FrameListener frameListener;
     private long gracefulShutdownTimeoutMillis = DEFAULT_GRACEFUL_SHUTDOWN_TIMEOUT_MILLIS;
+    private boolean gracefulShutdownIndefiniteWait;
```

Comment:
instead of adding extra state can we just allow `gracefulShutdownTimeoutMillis` to be negative and that represent infinite time?